### PR TITLE
Proper disconnect and stop cpu profiling

### PR DESCRIPTION
**Pull request by revathskumar**
**Review and merge only after #5**
- Now we are able to disconnect from both the queues
- Able to send stop_cpu_profiling properly.

I believe still need some refactoring on store side.

### CODE REVIEW

**Comment by revathskumar** [comment]
@emilsoman As per the last review I updated the PR with hardcoding localhost for connection and removed the textbox from connection form.

File: app/redux/reducers/connection.js [hunk 1]
```diff
@@ -30,11 +31,19 @@ const handshake = function (state = {}, action) {
...
       if (action.from !== 'subscriber') {
-        return state;
+        return Object.assign({}, state, {status: 'STOPPED'});
+      }
+      let data = state.data.splice(0);
+      if (state.status === 'STOPPED') {
+        data = [];
       }
-      return Object.assign({}, state, {status: 'RUNNING', data: [...state.data, action.data], from: action.from});
+      return Object.assign({}, state, {status: 'RUNNING', data: [...data, action.data], from: action.from});
```
Comment (kgrz) [inline app/redux/reducers/connection.js:45]: For future reference, naming something `data` can be pretty opaque.

Comment (revathskumar) [inline app/redux/reducers/connection.js:45]: @kgrz Noted.

File: app/components/toolbelt.js [hunk 2]
```diff
@@ -11,12 +11,16 @@ class Toolbelt extends Component {
...
-        <button className={`btn ${btnClass}`} data-status={cpuProfile.status} onClick={this.props.handleCpuSampling}>
-          {btnText}
-          {cpuProfile.status === 'RUNNING' ? <i className="fa fa-spinner fa-spin"></i> : null}
-        </button>
+      <div className="row">
+        <div className="col-md-2">
+          <Button value="Handshake" onClick={this.props.handshake}/>
+        </div>
+        <div className="col-md-3">
+          <button className={`btn ${btnClass}`} data-status={cpuProfile.status} onClick={this.props.handleCpuSampling}>
```
Comment (kgrz) [inline app/components/toolbelt.js:19]: This is minor, but I found returning the entire classname in the classname generating routine pretty helpful in making the JSX cleaner. For instance, the `btnClass` can hold `'btn btn-success` thereby removing the use of interpolation here.

Comment (revathskumar) [inline app/components/toolbelt.js:19]: This is fixed in 338d875e2447cbe07b4375210fe13053933665f7

File: app/components/layout.js [hunk 3]
```diff
@@ -40,17 +39,18 @@ class Layout extends React.Component {
...
+        <Navbar {...this.props} handleConnect={this.connect} handleDisconnect={this.disconnect}/>
         <div className="container">
-          <div className="starter-template">
-            <h1>Bootstrap starter template</h1>
-            <p className="lead">
-              Use this document as a way to quickly start any new project.
-            </p>
+          <div className="row tools-wrapper">
+            <Toolbelt {...this.props} handshake={this.handshake} handleCpuSampling={this.cpuProfiling}/>
+            <div className="row">
```
Comment (siddharthbhagwan) [inline app/components/layout.js:46]: Should we even show the toolbelt while the app isn't connected?

As per the app now, if it is disconnected and someone clicks the start profiling button, and then connects, the profiling starts post a connection establishment. I think that makes it a little complicated.

File: app/components/connectionForm.js [hunk 4]
```diff
@@ -1,36 +1,27 @@
...
+
+const Connect = props => {
+  return props.status === 'CONNECTED' ? <DisconnectButton {...props}/> : <ConnectButton {...props}/>;
+};
+
+Connect.propTypes = {
+  status: PropTypes.string
+};
+
+export default Connect;
```
Comment (siddharthbhagwan) [inline app/components/connectionForm.js:27]: Not sure what the convention is, is it a good practice to have stateless components within another component or should they be in separate files? On the other hand, these are most likely not going to be re used anywhere.

File: app/components/layout.js (hunk 3 above)
Comment (revathskumar) [inline app/components/layout.js:46]: ok. we can hide the tool belt when client is disconnected from sever.

**Comment by revathskumar** [comment]
If there is no other comments. Can I merge this? After merging this reviewing other PR might get easier. 
cc\ @emilsoman @kgrz

**Pull request merged**